miso-web/src/main/webapp/scripts/array_ajax.js

```diff
@@ -0,0 +1,389 @@
+
+    showSamplesLoading(true);
+
+    $.ajax({
+      url: '/miso/rest/arrays/' + arrayJson.id + '/positions/' + selectedPosition,
+      type: 'DELETE',
+      contentType: 'application/json; charset=utf8'
+    }).success(function(data) {
+      clearSampleSearchResults();
+      SampleArray.setArrayJson(data);
+    }).fail(function(response, textStatus, serverStatus) {
+      Utils.showOkDialog('Error removing sample', [JSON.parse(response.responseText).detail]);
+      showSamplesLoading(false);
+    });
+  };
+
+  SampleArray.searchSamples = function() {
+    var searchString = $('#searchField').val();
+    if (!searchString) {
+      clearSampleSearchResults();
+      return;
+    }
+    showSamplesLoading(true);
+    var url = "/miso/rest/arrays/sample-search?" + jQuery.param({
+      q: searchString
+    });
+    $.ajax({
+      url: url,
+      contentType: "application/json; charset=utf8",
+      type: "GET"
+    }).success(function(data) {
+      showSampleSearchResults(data);
+    }).fail(function(response, textStatus, serverStatus) {
+      clearSampleSearchResults();
+      var error = JSON.parse(response.responseText);
+      var message = error.detail ? error.detail : error.message;
+      Utils.showOkDialog('Search error', [message]);
+    });
+  };
+
+  SampleArray.updatePosition = function() {
+    if (visual.selectedItems.length !== 1) {
+      $('#warningMessages').html("Please select a single position from the grid, then retry.");
+      return;
+    } else if ($('#resultSelect').val() == -1) {
+      $('#warningMessages').html('Please select an item to add.');
+      return;
+    }
+
+    var selectedPosition = visual.selectedItems[0].position;
+    var selectedItem = getItemAtPosition(selectedPosition);
+
+    var addTheItem = function() {
+      showSamplesLoading(true);
+
+      $.ajax({
+        url: '/miso/rest/arrays/' + arrayJson.id + '/positions/' + selectedPosition + '?' + jQuery.param({
+          sampleId: jQuery('#resultSelect').val()
+        }),
+        type: "PUT",
+        contentType: 'application/json; charset=utf8'
+      }).success(function(data) {
+        clearSampleSearchResults();
+        SampleArray.setArrayJson(data);
+      }).fail(function(response, textStatus, serverStatus) {
+        var error = JSON.parse(response.responseText);
+        var message = error.detail ? error.detail : error.message;
+        $('#warningMessages').html('Error adding item: ' + message);
+        $('#ajaxLoader').addClass('hidden');
+        showSamplesLoading(false);
+      });
+    };
+
+    if (selectedItem) {
+      if (selectedItem.id == $('#resultSelect').val()) {
+        // setting same item where it already is. No change necessary
+        clearSampleSearchResults();
+        $('#searchField').val('');
+      } else {
+        // if selectedPosition is already filled, confirm before deleting that position
+        var sampleInfo = selectedItem.name + ' (' + selectedItem.alias + ')';
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "sampleInfo" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of the unused 'sampleInfo' variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AUnusedVariable)